runtime/src/main/scala/akka/grpc/scaladsl/RequestBuilder.scala

```diff
@@ -48,6 +49,12 @@ trait SingleResponseRequestBuilder[Req, Res] {
    * Invoke the gRPC method with the additional metadata added and provide access to response metadata
    */
   def invokeWithMetadata(request: Req): Future[GrpcSingleResponse[Res]]
+
+  /**
+   * Set the deadline for this call
+   * @return A new request builder, that will use the supplied deadline when invoked
+   */
+  def setDeadline(deadline: Duration): SingleResponseRequestBuilder[Req, Res]
```

Comment:
Should we use `FiniteDuration` here instead. Always somewhat confusing what non finite Duration means.

Reply:
I went to make this change but found that `GrpcClientSettings#deadline` is already typed as `Duration` and supports supplying infinite, which means no timeout -- in `NettyClientUtils` it's implemented as follows:

```
  @InternalApi private[akka] def callOptionsWithDeadline(
      defaultOptions: CallOptions,
      settings: GrpcClientSettings): CallOptions =
    settings.deadline match {
      case d: FiniteDuration => defaultOptions.withDeadlineAfter(d.toMillis, TimeUnit.MILLISECONDS)
      case _                 => defaultOptions
    }
```

...so, I updated the call sites to check `isFinite`/null (Scala/Java) and clear timeout if its infinite, which I think is more consistent with how the settings are parsed. Let me know what you think though - happy to make it `FiniteDuration` if you think that's better.

Reply:
good, thanks